### DISCUSSION TITLE
Run on CPU

### DISCUSSION THREAD

**Discussion by zzxxddffdle** [basujindal/stable-diffusion#151] [Q&A]
Is it possible to make this run on a CPU?

**Comment by bitRAKE** [comment]
There is indeed CPU support for Pytorch, but optimization on the CPU is a different landscape, imho. For example, maybe there isn't native FP16 support (i.e. AVX-512 on Intel/AMD) and it might be more practical to stay with FP32 (given the typically larger memory). For me, this seems outside the scope of this repo.

I do agree, having a device agnostic fork would be nice. Maybe, one already exists?

**Reply by bitRAKE** [reply]
https://github.com/basujindal/stable-diffusion/issues/130 seems like it already runs on CPU - guess I should try it before commenting.

**Reply by zzxxddffdle** [reply]
Although basujindal /stable-diffusion runs well on my 1060 at 512x512, I would like to use my pc's 32 gb to render larger images so it would be nice to run basujindal /stable-diffusion on the 7700k. I haven't a clue how, though and would need some step by step instructions.

**Reply by bitRAKE** [reply]
Tried it with the current repo and just adding ```--device cpu``` to the command line is sufficient. Pytorch doesn't appear to be using hyperthreading - which is what that issue is regarding.
